### PR TITLE
Add multi-process example

### PR DESCRIPTION
**Pull request by beeender**
All Realm public APIs should be process-safe now.

Encryption is not supported for multi processes due to core
restriction https://github.com/realm/realm-core/issues/1845

Accessing same Realm file in different processes from different apks
is safe, but the notifications won't work.

### CODE REVIEW

**Comment by Zhuinden** [comment]
> Accessing same Realm file in different processes from different apks
is safe, but the notifications won't work.

So if another APK writes into Realm, the other APK won't be notified? :confused:

**Comment by beeender** [comment]
> So if another APK writes into Realm, the other APK won't be notified? 😕

Yes. Mostly because of the SeLinux, we cannot create pipe on the external storage. But notifications rely on the pipe. It could be solved by adding a new public API to set up a realm temp directory to one of the apk's internal storage, but then the other apk won't work independently. Also, that requires both apk use the same uid by `sharedUserId` in manifest https://developer.android.com/guide/topics/manifest/manifest-element.html#uid

It will be quite fun to just use Realm to communicate between apks ...

**Comment by Zhuinden** [comment]
Understandable, I guess this really is an issue only if a Realm is exposed through a Content Provider and both APKs write into it simultaneously. Even then, you could manually broadcast change events through Content Provider, at least notifications work in the same APK

File: examples/multiProcessExample/build.gradle [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    productFlavors {
+    }
+
+    command {
+        monkey.events 2000
+    }
+}
+
+dependencies {
+    compile 'com.android.support:appcompat-v7:23.1.1'
```
Comment (zaki50) [inline examples/multiProcessExample/build.gradle]: We need to use the same major version of `appcompat` with `compileSdkVersion`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,8 @@
 
 ### Enhancements
 
+* Added support for using non-encrypted Realm in multiple processes from the same apk (#1091).
```
Comment (cmelchior) [inline CHANGELOG.md]: This is slightly pessimistic as using `sharedUserId` allows multiple APK's to access it, but I'm not entirely sure how to reword it to make it clear since there is also a bunch of other caveats. Perhaps something likes

```
Added support for using non-encrypted Realms in multiple processes. Some caveats apply. Read <insertLinkToDocs> for more info (#1091).
```

File: examples/settings.gradle [hunk 3]
```diff
@@ -13,6 +13,7 @@ include 'unitTestExample'
 include 'newsreaderExample'
 include 'rxJavaExample'
 include 'objectServerExample'
+include 'multiProcessExample'
```
Comment (cmelchior) [inline examples/settings.gradle]: Multiprocess is one word, so this should be `multiprocessExample`

File: examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/AnotherProcessService.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
+package io.realm.examples.realmmultiprocessexample;
```
Comment (cmelchior) [inline examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/AnotherProcessService.java]: Missing license in this file and a bunch of others

File: examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2014 Realm Inc.
```
Comment (cmelchior) [inline examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/MainActivity.java]: 2017 in this file and a bunch of others

File: examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        @Override
+        public void onChange(RealmResults<ProcessInfo> results) {
+            StringBuilder stringBuilder = new StringBuilder();
+
+            for (ProcessInfo processInfo : results) {
+                stringBuilder.append(processInfo.getName());
+                stringBuilder.append("\npid: ");
+                stringBuilder.append(processInfo.getPid());
+                stringBuilder.append("\nlast response time: ");
+                DateFormat df = new SimpleDateFormat("MM/dd/yyyy HH:mm:ss", Locale.ENGLISH);
```
Comment (cmelchior) [inline examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/MainActivity.java]: Move this to a class variable?

File: examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/Utils.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    public static String getMyProcessName(Context context) {
+        String processName = "";
+        ActivityManager am = (ActivityManager)context.getSystemService(Context.ACTIVITY_SERVICE);
+        List<RunningAppProcessInfo> infoList= am.getRunningAppProcesses();
+        if (infoList == null) {
+            throw new RuntimeException("getRunningAppProcesses() returns 'null'.");
+        }
+        for (RunningAppProcessInfo info : infoList) {
+            try {
+                if (info.pid == Process.myPid()) {
```
Comment (cmelchior) [inline examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/Utils.java]: Can this throw?

File: examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/MainActivity.java (hunk 5 above)
Comment (beeender) [reply examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/MainActivity.java]: UH, yeah! This code was written in 2014 actually ! will change it.

File: examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/Utils.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.List;
+
+import io.realm.examples.realmmultiprocessexample.models.ProcessInfo;
+
+public class Utils {
+
+    public static String getMyProcessName(Context context) {
+        String processName = "";
+        ActivityManager am = (ActivityManager)context.getSystemService(Context.ACTIVITY_SERVICE);
+        List<RunningAppProcessInfo> infoList= am.getRunningAppProcesses();
```
Comment (nhachicha) [inline examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/Utils.java]: `infoList=` missing space

File: examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/Utils.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        if (infoList == null) {
+            throw new RuntimeException("getRunningAppProcesses() returns 'null'.");
+        }
+        for (RunningAppProcessInfo info : infoList) {
+            try {
+                if (info.pid == Process.myPid()) {
+                    processName = info.processName;
```
Comment (nhachicha) [inline examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/Utils.java]: shouldn't you break from the for loop if you find your process name?

File: examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/models/ProcessInfo.java [hunk 10]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2014 Realm Inc.
```
Comment (nhachicha) [inline examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/models/ProcessInfo.java]: update year

File: examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/Utils.java (hunk 7 above)
Comment (beeender) [reply examples/multiProcessExample/src/main/java/io/realm/examples/realmmultiprocessexample/Utils.java]: nope, should never throw. Unless it is called in a terminated process :P
